src/index.ts

```
@@ -1,32 +1,37 @@
-class User {
+interface IUser {
+  id?: number
   name: string
   favColor: string
   age: number
-    
-  constructor(name: string, color: string, age: number) {
-    this.name = name
-    this.favColor = color
-    this.age = age
-  }
 }
 
 class UserAPI {
-  users: Array<User>
+  private users: Array<IUser>
+  private nextId: number
 
   constructor() {
     this.users = []
+    this.nextId = 0
   } 
 
   private assignId() {
-
+    this.nextId++
+    return this.nextId
   }
 
-  addUser(user: User) {
-
+  addUser(user: IUser) {
+    const {name, favColor, age} = user
```

if the user has an id already, this function should throw an exception (see javascript.info for how to throw errors)

---

Can I get some clarifications here: 
1) Will the user object being passed to the API have a pre-existing ID property that should be store in the database?
1a) If so, the error checking to to verify no two user have the same ID, correct? 
2) You want error checking for duplicate users. ex name: {"andy" age: 111}  => Error: User already exists in the database

---

`Will the user object being passed to the API have a pre-existing ID property that should be stored in the database?` someone calling the addUser method could formulate a user object with the id already set. This is an error condition, becuase we want the userapi to be the one assigning ids.

`You want error checking for duplicate users. ex name: {"andy" age: 111} => Error: User already exists in the database` correct

---

Thank you